src/qcodeedit/lib/qeditor.cpp

```diff
@@ -3072,14 +3072,17 @@ void QEditor::selectOccurence(bool backward, bool keepMirrors, bool all)
 void QEditor::setVerticalScrollBarMaximum()
 {
 	if (!m_doc) return;
-	const QSize viewportSize = viewport()->size();
-	int viewportHeight = viewportSize.height();
-	if (flag(VerticalOverScroll))
-		viewportHeight /= 2;
-    const qreal ls = m_doc->getLineSpacing();
+	const qreal ls = m_doc->getLineSpacing();
+	const int totalLines = qRound(m_doc->height() / ls);

```

Comment:
it should probably not be qRound.
qCeil ?

Reply:
definitely different from old, so please use that (or functional **identical** )

Reply:
from a numerical point of view height() = ls * totalLines. Due to rounding errors calculating totalLines from this equation will likely not give an exact integer but a real that is very close to an integer. qRound would fix this. Adding 0.5 and dropping the fractional part (convertion to int) should do the same. But original solution adds 1. for some reason I don't understand. I revert this.

Reply:
Observations:
1. ``m_doc->height()`` and ``ls`` have fractions. But interesting enough the result of ``m_doc->height() / ls`` is a real that represents an exact integer. Let x be the result, then the following shows that x, qFloor(x), and qCeil(x) are the same (i.e. x is an integer) even with more than 786 thousand lines. This is independent of ``ls`` being 1, 1.25, or 17.4375.

    Examples:
    ```
    ls: 10.75  height() : 494.5  height()/ls : 46 46 46

    ls: 53.6875  height() : 2469.63  height()/ls : 46 46 46

    ls: 1.34375  height() : 1.05677e+06  height()/ls : 786433 786433 786433

    ls: 17.4375  height() : 1.37134e+07  height()/ls : 786433 786433 786433
    ```
    The exact calculation of the last line for overscrolling to the top ``totalLines = qRound(m_doc->height() / ls)`` does not rely totally on this observation by using qRound.

3. ``viewport()->size().height()`` is an integer (pixels), but dividing it by ``ls`` gives a real:
    ```
    ls: 17.4375  height(): 479  height() / ls: 27.4695
    ```
    The integer part of this result is used for pageStep.